projects/storefrontlib/src/cms-components/asm/asm-enabler.service.spec.ts

```diff
@@ -0,0 +1,119 @@
+  path() {
+    return '';
+  }
+}
+
+describe('AsmEnablerService', () => {
+  let asmEnablerService: AsmEnablerService;
+  let windowRef: WindowRef;
+  let location: Location;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [
+        { provide: WindowRef, useValue: MockWindowRef },
+        {
+          provide: ComponentFactoryResolver,
+          useClass: MockComponentFactoryResolver,
+        },
+        { provide: OutletService, useClass: MockOutletService },
+        { provide: Location, useClass: MockLocation },
+      ],
+    });
+
+    asmEnablerService = TestBed.get(AsmEnablerService);
+    windowRef = TestBed.get(WindowRef);
+    location = TestBed.get(Location);
+
+    windowRef.localStorage.removeItem(BROWSER_STORAGE_KEY);
+  });
+
+  it('should be created', () => {
+    expect(asmEnablerService).toBeTruthy();
+  });
+
+  describe('Open ASM based on URL parameter', () => {
+    it('should add UI when ?asm=true', () => {
+      spyOn(location, 'path').and.returnValue('/any/url?asm=true');
+      spyOn(<any>asmEnablerService, 'addUi').and.stub();
+      asmEnablerService.load();
+      expect((<any>asmEnablerService).addUi).toHaveBeenCalled();
+    });
+
+    it('should not add UI when asm param is not used', () => {
+      spyOn(location, 'path').and.returnValue('/any/url');
+      spyOn(<any>asmEnablerService, 'addUi').and.stub();
+      asmEnablerService.load();
+      expect((<any>asmEnablerService).addUi).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('Open ASM based on previous usage', () => {
+    it('should add UI when localStorage key asm_enabled is true', () => {
+      windowRef.localStorage.setItem(BROWSER_STORAGE_KEY, 'true');
+      spyOn(location, 'path').and.returnValue('/any/url');
+      spyOn(<any>asmEnablerService, 'addUi').and.stub();
+      asmEnablerService.load();
+      expect((<any>asmEnablerService).addUi).toHaveBeenCalled();
+    });
+
+    it('should not add UI when localStorage asm_enabled is false ', () => {
+      windowRef.localStorage.setItem(BROWSER_STORAGE_KEY, 'false');
+      spyOn(location, 'path').and.returnValue('/any/url');
+      spyOn(<any>asmEnablerService, 'addUi').and.stub();
+      asmEnablerService.load();
+      expect((<any>asmEnablerService).addUi).not.toHaveBeenCalled();
+    });
+
+    it('should not add UI when localStorage asm_enabled is not available ', () => {
+      windowRef.localStorage.removeItem(BROWSER_STORAGE_KEY);
+      spyOn(location, 'path').and.returnValue('/any/url');
+      spyOn(<any>asmEnablerService, 'addUi').and.stub();
+      asmEnablerService.load();
+      expect((<any>asmEnablerService).addUi).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('Unload', () => {
+    it('should remove local storage key to false on unload', () => {
+      windowRef.localStorage.setItem(BROWSER_STORAGE_KEY, 'true');
+      asmEnablerService.unload();
+      expect(windowRef.localStorage.getItem(BROWSER_STORAGE_KEY)).toBeNull();
```

Comment:
In the browser, I observe that the key is still there when we unload Asm.  The value is false.

Reply:
yes, while making tests, I learned to remove the storage instead. Good point.